### PR TITLE
Create scalar transport module

### LINKED ISSUE #13192: Scalar Transport Module
## Reason
At a minimum this module would serve as a good tutorial for concepts like hyperbolic PDEs and methods for solving them, e.g. SUPG, DG, RDG.

## Design
Following up on the reason above, this should pull in the RDG module. We should be able to use the Scalar Transport module to reproduce the results shown in some of @yidongxiainl's test problems [here](https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!searchin/moose-users/yidong$20xia$20discontinuous$20galerkin%7Csort:date/moose-users/p6h7PqZNOrA/sNmPZiDBumkJ).

## Impact
MOOSE is a general multi-physics tool and I think it's an excellent platform for exploring and teaching concepts of FEM. Showing that we can do different transport methods and illustrate the results expected from literature is a good thing in my book!

### PR DESCRIPTION
**Pull request by lindsayad**
Closes #13192

### CODE REVIEW

**Comment by lindsayad** [comment]
Thanks a ton @cticenhour!

**Comment by cticenhour** [comment]
No problem! I just now noticed I need to make one more tweak. Doesn't impact SQA docs building, but does change how it gets rendered. Will fixup my last commit real quick.

File: modules/scalar_transport/src/kernels/TimeDerivativeLM.C
Comment (loganharbour) [inline modules/scalar_transport/src/kernels/TimeDerivativeLM.C:11]: Unnecessary include I think

**Review by loganharbour — CHANGES_REQUESTED** [review]

File: modules/scalar_transport/doc/moosedocs.py
Comment (lindsayad) [reply modules/scalar_transport/doc/moosedocs.py:1]: I did

File: modules/scalar_transport/include/kernels/BodyForceLM.h
Comment (lindsayad) [reply modules/scalar_transport/include/kernels/BodyForceLM.h:7]: Ugh I got the source files ... how did I miss the headers!

**Comment by lindsayad** [comment]
@cticenhour I need to add a dependency for the `misc` module for documentation. I tried to follow what is done in `thermal_hydraulics` but that lead me to errors like
```
MooseDocs.base.readers (ForkProcess-15): 
TOKENIZE ERROR: '/Materials/WallHeatTransferCoefficient3EqnDittusBoelterMaterial' syntax was not recognized.
/home/lindad/projects/moose/modules/thermal_hydraulics/doc/content/modules/thermal_hydraulics/source/materials/WallHeatTransferCoefficient3EqnDittusBoelterMaterial.md:12
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
12│                                                                                                │
13│                                                                                                │
14│!syntax inputs /Materials/WallHeatTransferCoefficient3EqnDittusBoelterMaterial                  │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
```

**Comment by GiudGiud** [comment]
we ll need the TMAP issues for these objects moved to MOOSE?

**Comment by lindsayad** [comment]
> we ll need the TMAP issues for these objects moved to MOOSE?

I would need an SQA expert to tell me if that's necessary. Unlike for THM the TMAP8 repository is not being archived. It's issue tracker is still active

**Comment by cticenhour** [comment]
> Unlike for THM the TMAP8 repository is not being archived. It's issue tracker is still active

A couple things here. There is the capability in MooseDocs to refer to another repo's issues for tracking purposes. We would need to make a small change to the `moose/modules/doc/config.yml` to allow for `TMAP8#1234` in an `issues` parameter for a test spec. 

That establishes what we *can* do. Now whether that's something we *want* to do is another question. My opinion is that TMAP8 would likely then need to be subject to some measure of INL-based backup and archival alongside MOOSE if we want to start referring to TMAP8 issues (even if we never do it again beyond this point). We can't lose linked issues in our requirements listing if something catastrophic were to happen and we lose things on GitHub. To my knowledge, MOOSE is the only public repo being actively backed up along with its issues (because we're required to do so). 

So, I would say we do need to duplicate some issues (with GitHub links to the original TMAP8 issues for reference) in MOOSE to reflect the prompting issue (and containing all relevant context) for each of the tests/requirements being added. @permcody might also want to chime in with his own opinion.

**Comment by lindsayad** [comment]
Can we selectively apply whatever issue migration script was used for thermal hydraulics?

**Comment by loganharbour** [comment]
> Can we selectively apply whatever issue migration script was used for thermal hydraulics? Only would currently need to apply it for [idaholab/TMAP8#13](https://github.com/idaholab/TMAP8/issues/13) and [idaholab/TMAP8#14](https://github.com/idaholab/TMAP8/issues/14)

Yeah, we can. I'm the one that did the transfer and still have it available. Let me know

**Comment by lindsayad** [comment]
Yea let's do it. Just those two issues, at this time at least

File: modules/modules.mk
Comment (loganharbour) [inline modules/modules.mk:107]:
```suggestion
	      CHEMICAL_REACTIONS          := yes
	      NAVIER_STOKES               := yes
	      THERMAL_HYDRAULICS	        := yes
        FLUID_PROPERTIES            := yes
        HEAT_CONDUCTION             := yes
        RDG                         := yes
        RAY_TRACING                 := yes
        SOLID_PROPERTIES            := yes
        MISC                        := yes
```

File: modules/scalar_transport/doc/content/source/bcs/BinaryRecombinationBC.md
Comment (loganharbour) [inline modules/scalar_transport/doc/content/source/bcs/BinaryRecombinationBC.md:13]:
```suggestion
where $K_r$ is the recombination coefficient specified with [!param](/BCs/BinaryRecombinationBC/Kr),
```

File: modules/scalar_transport/doc/content/source/bcs/BinaryRecombinationBC.md
Comment (loganharbour) [inline modules/scalar_transport/doc/content/source/bcs/BinaryRecombinationBC.md:15]:
```suggestion
$u$ corresponds to [!param](/BCs/BinaryRecombinationBC/variable) and $v$ is a coupled variable specified by [!param](/BCs/BinaryRecombinationBC/v).
As the name states, this class is meant to model binary recombination
```

File: modules/scalar_transport/doc/content/source/bcs/DissociationFluxBC.md
Comment (loganharbour) [inline modules/scalar_transport/doc/content/source/bcs/DissociationFluxBC.md:16]:
```suggestion
where $K_d$ is a dissociation coefficient specified with [!param](/BCs/DissociationFluxBC/Kd) and
$v$ is the coupled variable specified with [!param](/BCs/DissociationFluxBC/v) that is
dissociating at the boundary to create the species corresponding to
[!param](/BCs/DissociationFluxBC/variable).
```

File: modules/scalar_transport/doc/content/source/bcs/UnaryRecombinationBC.md
Comment (loganharbour) [inline modules/scalar_transport/doc/content/source/bcs/UnaryRecombinationBC.md:14]:
```suggestion
where $K_r$ is the recombination coefficient specified with [!param](/BCs/UnaryRecombinationBC/Kr)
and $u$ corresponds to [!param](/BCs/UnaryRecombinationBC/variable).
```

**Review by loganharbour — CHANGES_REQUESTED** [review]
Good to go after this; you should be able to batch these changes

File: modules/modules.mk
Comment (cticenhour) [reply modules/modules.mk:107]: What's with all the extra spaces here?

Comment (cticenhour) [reply modules/modules.mk:107]: I see now - weird how the rendering is different in the snippet view compared to the overview.

File: modules/modules.mk
Comment (GiudGiud) [inline modules/modules.mk:37]: same for spaces here

File: modules/modules.mk
Comment (GiudGiud) [inline modules/modules.mk:98]: and here

File: modules/scalar_transport/include/bcs/BinaryRecombinationBC.h
Comment (GiudGiud) [inline modules/scalar_transport/include/bcs/BinaryRecombinationBC.h:28]:
```suggestion
  /// The concentration of B, e.g. the concentration of the other species recombining with the
```

File: modules/scalar_transport/include/kernels/CoupledForceLM.h
Comment (GiudGiud) [inline modules/scalar_transport/include/kernels/CoupledForceLM.h:14]: docstring

File: modules/scalar_transport/include/kernels/LMKernel.h
Comment (GiudGiud) [inline modules/scalar_transport/include/kernels/LMKernel.h:14]: docstrings

File: modules/scalar_transport/include/kernels/LMTimeKernel.h
Comment (GiudGiud) [inline modules/scalar_transport/include/kernels/LMTimeKernel.h:12]: docstrings

File: modules/scalar_transport/src/bcs/BinaryRecombinationBC.C
Comment (GiudGiud) [inline modules/scalar_transport/src/bcs/BinaryRecombinationBC.C:18]:
```suggestion
  params.addClassDescription("Models recombination of the variable with a coupled species at "
```

File: modules/scalar_transport/src/bcs/UnaryRecombinationBC.C
Comment (GiudGiud) [inline modules/scalar_transport/src/bcs/UnaryRecombinationBC.C:17]: do we really need that? it s the same as binary with the same parameter

File: modules/scalar_transport/test/tests/ncp-lms/README.md
Comment (GiudGiud) [inline modules/scalar_transport/test/tests/ncp-lms/README.md:1]: this file should be used in some capacity. looks like it's just there? is it linked in the docs?

File: modules/scalar_transport/test/tests/ncp-lms/elemental.cmp
Comment (GiudGiud) [inline modules/scalar_transport/test/tests/ncp-lms/elemental.cmp:4]: not used?

**Review by GiudGiud — CHANGES_REQUESTED** [review]

**Comment by loganharbour** [comment]
In comes the triple review with @GiudGiud... sorry @lindsayad 

ping me before merging and I'll get civet stuff ready

File: modules/scalar_transport/test/tests/ncp-lms/README.md
Comment (loganharbour) [reply modules/scalar_transport/test/tests/ncp-lms/README.md:1]: It's a detailed description of the tests in the form of a markdown, not in official docs. this is fine. it can be used as a starting point for detailed doco in the future.

File: modules/scalar_transport/src/bcs/UnaryRecombinationBC.C
Comment (lindsayad) [reply modules/scalar_transport/src/bcs/UnaryRecombinationBC.C:17]: @loganharbour what do you mean by the 👎 ?

Comment (loganharbour) [reply modules/scalar_transport/src/bcs/UnaryRecombinationBC.C:17]: I mean that I see the difference and am fine with the two objects

**Review by GiudGiud — APPROVED** [review]

**Comment by lindsayad** [comment]
@loganharbour looks like we're about ready

**Review by loganharbour — APPROVED** [review]

**Pull request merged**
